UefiTestingPkg/FunctionalSystemTests/MemoryProtectionTest/App/MemoryProtectionTestApp.c

```diff
@@ -618,7 +644,10 @@ ImageProtectionPreReq (
   IN UNIT_TEST_CONTEXT  Context
   )
 {
-  if (mDxeMps.ImageProtectionPolicy.Fields.ProtectImageFromFv || mDxeMps.ImageProtectionPolicy.Fields.ProtectImageFromUnknown) {
+  if (((mDxeMps.ImageProtectionPolicy.Fields.ProtectImageFromFv != 0) || (mDxeMps.ImageProtectionPolicy.Fields.ProtectImageFromUnknown != 0)) &&
+      ((mMemoryAttributeProtocol != NULL) || !EFI_ERROR (PopulateMemoryAttributeProtocol ())) &&

```

Comment:
Can this be simplified by dropping the global variable checks and let them be checked in the functions - `mMemoryAttributeProtocol` in `PopulateMemoryAttributeProtocol` and `mMemoryProtectionProtocol` in `PopulateMemoryProtectionDebugProtocol`?

Reply:
Sure, removed